base_external_dbsource/models/base_external_dbsource.py

```diff
@@ -296,19 +285,6 @@ def _execute_generic(self, query, params, metadata):
 
     # Compatibility & Private
 
-    def conn_open(self):
-        """It opens and returns a connection to the remote data source.
-
-        This method calls adapter method of this same name, suffixed with
-        the adapter type.
-
-        Deprecate:
-            This method has been replaced with ``connection_open``.
-        """
-
-        with self.connection_open() as connection:
-            return connection
-
     def _get_adapter_method(self, method_prefix):
```

Comment:
This suggestion makes total sense. 

However, the current system doesn't make end uses of the base module more complex (because all the `_get_adapter_method()` dance is done directly in the base module, and only there), and it doesn't make sub-implementations particularly complex (see [this example](https://github.com/OCA/server-backend/blob/427c6cccc84d605cec2f8a5b115f68807cecc292/base_external_dbsource_sqlite/models/base_external_dbsource.py); it's quite simple anyway).

In this refactor, I'm changing stuff that actually cause problems. But following this suggestion would make it a deeper change for something that, although not very ergonomic, isn't causing problems. Thus, I think it's better to keep it as it is, to minimize the refactor here.